.github/workflows/staging.yml

```diff
@@ -0,0 +1,74 @@
+name: Build and Deploy Prod
+
+on:
+  push:
+    branches:
+    - 'feature/rebranding'
+
+jobs:
+  docker-build:
+    runs-on: ubuntu-latest
+    outputs:
+      image-version: ghcr.io/archway-network/archway-docs:${{ steps.meta.outputs.version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          # TODO: ghrc credentials need to be moved to a shared infra
+          # account
+          username: 'shahbazn'
+          password: ${{ secrets.GHCR_ACCESS_TOKEN }}
+
+      - name: Set Docker Metadata
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/archway-network/archway-docs
+          tags: |
+            type=semver,pattern={{raw}}
+            type=sha
+
+      - name: Build and push docker image
+        uses: docker/build-push-action@v4
+        with:
+          push: false
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          secrets: |
+            ALGOLIA_APPLICATION_ID=${{ secrets.ALGOLIA_APPLICATION_ID }}
+            ALGOLIA_SEARCH_API_KEY=${{ secrets.ALGOLIA_SEARCH_API_KEY }}
+            ALGOLIA_WRITE_API_KEY=${{ secrets.ALGOLIA_WRITE_API_KEY }}
+            ALGOLIA_INDEX_STAGING=${{ secrets.ALGOLIA_INDEX_STAGING }}
```

Comment:
this should be ALGOLIA_INDEX... the build step should use github environments

For example:

jobs:
  build:
    uses: ./.github/workflows/docker-build.yml
    with:
      environment: production
    secrets: inherit